changelog/8335.improvement.rst

```diff
@@ -0,0 +1,10 @@
+Improved pytest.approx assertion messages
+
+The assertion messages now dumps a table with the index and the error of each diff.
+Example:
+
+    >       assert [1, 2, 3, 4] == pytest.approx([1, 3, 3, 5])
+    E       assert comparison failed for 2 values:
+    E         Index | Obtained | Expected
+    E         1     | 2        | 3 ± 3.0e-06
```

Comment:
Unfortunately the `±` breaks for me on Windows. I suggest we replace it with `+-` otherwise the docs won't build locally on Windows, although this doesn't seem to be a problem on Linux.

```suggestion
    E         1     | 2        | 3 +- 3.0e-06
```

Reply:
I find it weird, because this symbol was already being used on the previous assertion message format.
Should I change anyway? It'll "break" the convention with the previous implemented message, I believe ([See this](https://github.com/pytest-dev/pytest/blob/main/src/_pytest/python_api.py#L235))

Reply:
Ah, you meant only for the docs. Sorry. Will do.

Reply:
Exactly, sorry I should have been more explicit. 😁